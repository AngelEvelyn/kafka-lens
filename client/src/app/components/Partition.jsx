import PropTypes from 'prop-types';
import React from 'react';
import '../css/Partition.scss';

const Partition = ({ id, showMessages, topicName }) => (
  <div
    id={id}
    className="single-partition"
    role="button"
    tabIndex="0"
    onKeyPress={showMessages}
    onClick={showMessages}
    topicname={topicName}
  >
    {`Partition  ${id}`}
  </div>
);

export default Partition;

Partition.propTypes = {
  id: PropTypes.number.isRequired,
  showMessages: PropTypes.func.isRequired,
  topicName: PropTypes.string.isRequired,
};
