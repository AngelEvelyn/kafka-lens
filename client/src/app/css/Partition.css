.partition {
  border: 1px solid #ccc;
  border-radius: 2px;
  margin-bottom: 5px;
}

.partition-button {
}
