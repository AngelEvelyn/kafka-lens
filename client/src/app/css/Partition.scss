@import 'theme';

.single-partition {
    margin-top: 4px;
    padding-top: 3px;
    padding-bottom: 3px;
    padding-left: 8px;
    padding-right: 8px;
    border-radius: 3px;
    font-size: 16px;
}

.single-partition:hover {
    background-color: $highlight-bar-color;
    color: white;
    cursor: pointer;
}

.single-partition:focus {
    background-color: black;
}

