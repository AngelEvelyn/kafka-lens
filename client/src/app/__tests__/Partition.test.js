import React from 'react';
import { shallow } from 'enzyme';

import Partition from '../components/Partition';

describe('Parition.js unit tests', () => {
  let wrapper;

  const props = {
    id: 1
  };

  beforeAll(() => {
    wrapper = shallow(<Partition {...props} />);
  });

  it('Should render without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render a div with inner text equal to Parition', () => {
    expect(wrapper.find('div').text()).toEqual('Partition  1');
  });
});
