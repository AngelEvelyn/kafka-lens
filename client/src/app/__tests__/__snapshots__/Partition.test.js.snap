// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parition.js unit tests Should render without errors 1`] = `
<div
  className="single-partition"
  id={1}
>
  Partition  1
</div>
`;
